src/Symfony/Component/Validator/Constraints/ZeroComparisonConstraintTrait.php

```diff
@@ -21,9 +22,12 @@
  */
 trait ZeroComparisonConstraintTrait
 {
+    #[HasNamedArguments]
     public function __construct(?array $options = null, ?string $message = null, ?array $groups = null, mixed $payload = null)
     {
-        $options ??= [];
```

Comment:
this should probably be kept (after the deprecation) to avoid doing array access on `null`

Reply:
We must not enforce the array here to prevent a deprecation from being triggered in `AbstractComparison`. I added `is_array()` checks where necessary instead.